src/devices/ExplorerHat/Lighting/Lights.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+
+namespace Iot.Device.ExplorerHat.Lighting
+{
+    /// <summary>
+    /// Represents the Explorer HAT led array
+    /// </summary>
+    public class Lights : IDisposable
+    {
+        private const int LED1_PIN = 4;
+        private const int LED2_PIN = 17;
+        private const int LED3_PIN = 27;
+        private const int LED4_PIN = 5;
+
+        private GpioController _controller;
+
+        private List<Led> LedArray { get; set; }
+
+        /// <summary>
+        /// Blue led (#1)
+        /// </summary>
+        public Led One { get => LedArray[0]; }
+
+        /// <summary>
+        /// Yellow led (#2)
+        /// </summary>
+        public Led Two { get => LedArray[1]; }
+
+        /// <summary>
+        /// Red led (#3)
+        /// </summary>
+        public Led Three { get => LedArray[2]; }
+
+        /// <summary>
+        /// Green led (#4)
+        /// </summary>
+        public Led Four { get => LedArray[3]; }
+
+        /// <summary>
+        /// Blue led (#1)
+        /// </summary>
+        public Led Blue { get => LedArray[0]; }
+
+        /// <summary>
+        /// Yellow led (#2)
+        /// </summary>
+        public Led Yellow { get => LedArray[1]; }
+
+        /// <summary>
+        /// Red led (#3)
+        /// </summary>
+        public Led Red { get => LedArray[2]; }
+
+        /// <summary>
+        /// Green led (#4)
+        /// </summary>
+        public Led Green { get => LedArray[3]; }
+
+        /// <summary>
+        /// Gets the <see cref="Led"/> at the specified index
+        /// </summary>
+        /// <param name="key">The zero-based (0 to 3) of the led to get</param>
+        /// <returns>The <see cref="Led"/> at the specified index</returns>
+        public Led this[int key]
+        {
+            get
+            {
+                if (key < 0 || key > 3)
+                {
+                    throw new Exception("Leds are 0..3");
+                }
+
+                return LedArray[key];
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Led"/> at the specified index
+        /// </summary>
+        /// <param name="key">The color-string-index (blue, yellow, red or green) of the led to get</param>
+        /// <returns>The <see cref="Led"/> at the specified index</returns>
+        public Led this[string key]
```

Comment:
you should stick with one way of accessing LEDs, either property per LED or indexer or indexer by string but not all of them since this usually confuses users which one they should use

Reply:
You are right. I'll remove access by indexers.